persist/sqldb/archived_workflow_labels.go

```diff
@@ -8,8 +8,58 @@ import (
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/selection"
 	"upper.io/db.v3"
+
+	wfv1 "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1"
 )
 
+func (r *workflowArchive) ListWorkflowsLabelKeys() (*wfv1.LabelKeys, error) {
+	var archivedWfLabels []archivedWorkflowLabelRecord
+
+	err := r.session.
```

Comment:
I'd like us to reason about the SQL query plans that would be executed here. Are you able to write down both the actual SQL that would be executed, and point to which indexes would be used by that query?